### PR TITLE
Fix sdp bad constraint handling

### PR DESCRIPTION
**Pull request by trigaut**
Modify the way constraints are handled in SDP.
It is not necessary anymore to add the state constraints to the constraints function.
This PR should be merged quickly as the previous implementation will create issues in problem instances with constraints mixing states and controls

### CODE REVIEW

File: src/SDPoptimize.jl [hunk 1]
```diff
@@ -418,7 +413,15 @@ function get_control(model::SPModel,param::SDPparameters,V, t::Int64, x::Array)
...
 
-            if SDPmodel.constraints(t, next_state, u, w_sample)
+            next_state_box_const = true
+
+            for i in 1:model.dimStates
+                next_state_box_const =  (next_state_box_const&&
+                                        (next_state[i]>=model.xlim[i][1])&&
+                                        (next_state[i]<=model.xlim[i][2]))
+            end
+
```
Comment (frapac) [inline src/SDPoptimize.jl]: Could you DRY this pattern (as you use it more than once)?

File: examples/battery_storage_parallel.jl [hunk 2]
```diff
@@ -77,7 +77,7 @@ println("library loaded")
     end
 
     function constraint(t, x, u, xi)
-    	return( (x[1] <= s_bounds[1][2] )&(x[1] >= s_bounds[1][1]))
+    	return true
```
Comment (frapac) [inline examples/battery_storage_parallel.jl:80]: Could it be better to define a default function constraints(t, x, u ,xi) which is defined in the constructor of StochDynProgModel unless explicitly defined by the user?

Comment (trigaut) [inline examples/battery_storage_parallel.jl:80]: Yeah that's kind of what we do if we convert a LinearCostSPModel into StochDynProgModel. I could overload again the constructor but maybe it's time to harmonize the SPmodels constructor as there are almost identical. We already talked about that

File: src/SDPoptimize.jl (hunk 1 above)
Comment (trigaut) [inline src/SDPoptimize.jl]: I will put it in SDPutils ok

**Comment by frapac** [comment]
Ok to merge

**Pull request merged**